test/emulator_backend.go

```diff
@@ -493,31 +500,21 @@ func (e *EmulatorBackend) DeployContract(
 		return result.Error
 	}
 
-	return e.CommitBlock()
-}
-
-func (e *EmulatorBackend) ReadFile(path string) (string, error) {
-	// These are the scripts/transactions used by the
-	// BlockchainHelpers file.
-	if strings.HasPrefix(path, helperFilePrefix) {
-		filename := strings.TrimPrefix(path, helperFilePrefix)
-		switch filename {
-		case "mint_flow.cdc":
-			return string(MintFlowTransaction), nil
-		case "get_flow_balance.cdc":
-			return string(GetFlowBalance), nil
-		case "get_current_block_height.cdc":
-			return string(GetCurrentBlockHeight), nil
-		case "burn_flow.cdc":
-			return string(BurnFlow), nil
+	argTypes := make([]sema.Type, 0)
+	for _, arg := range args {
+		staticType := arg.StaticType(inter)
+		argType, err := inter.ConvertStaticToSemaType(staticType)
+		if err != nil {
+			panic(err)
 		}
+		argTypes = append(argTypes, argType)
 	}
-
-	if e.fileResolver == nil {
-		return "", FileResolverNotProvidedError{}
+	ContractInvocations[name] = DeployedContractConstructorInvocation{
```

Comment:
Don't think contract-name as the key would work - e.g. deploying the same contract two two different blockchains with different args.

Reply:
Hmm, but underlying they share the same source code. Does this affect the types defined in the contract? I know that the contracts will differ in their args, but their types remain the same. That was my reasoning behind this.

Reply:
I will try to write a test case for this though, out of curiosity.

Reply:
Yeah, that would not affect the nested types. Only affects if they call functions of the contract.

Reply:
I have added a relevant test case here: https://github.com/onflow/cadence-tools/pull/205/commits/aebb4e8590dac7eb6cfe8fee5bdbc734cf63f0b1